src/config/sentryConfig.ts

```diff
@@ -0,0 +1,174 @@
+import * as Sentry from "@sentry/react-native";
+import { logger } from "config/logger";
+import { useAnalyticsStore } from "ducks/analytics";
+import { useAuthenticationStore } from "ducks/auth";
+import { useNetworkStore } from "ducks/networkInfo";
+import { Platform } from "react-native";
+import Config from "react-native-config";
+import {
+  getVersion,
+  getBuildNumber,
+  getBundleId,
+} from "react-native-device-info";
+import { getUserId } from "services/analytics/user";
+
+/**
+ * Sentry configuration constants
+ */
+export const SENTRY_CONFIG = {
+  DSN: Config.SENTRY_DSN,
+
+  // Even when analytics are disabled, we still track errors but with minimal context
+  TRACK_ERRORS_WHEN_ANALYTICS_DISABLED: true,
+
+  // Reduced context when user has disabled analytics
+  MINIMAL_CONTEXT_FIELDS: [
+    "platform",
+    "platformVersion",
+    "network",
+    "appVersion",
+    "buildVersion",
+  ] as const,
+} as const;
+
+/**
+ * Builds common context data for Sentry events (similar to analytics).
+ *
+ * When analytics are enabled: Full context including connectivity info and public key
+ * When analytics are disabled: Minimal context for debugging without tracking user behavior
+ */
+const buildSentryContext = (
+  respectAnalyticsPreference = true,
+): Record<string, unknown> => {
+  const { isEnabled: analyticsEnabled } = useAnalyticsStore.getState();
+  const { connectionType, effectiveType } = useNetworkStore.getState();
+  const { network, account } = useAuthenticationStore.getState();
+
+  // Base context that's always included
+  const baseContext: Record<string, unknown> = {
+    platform: Platform.OS,
+    platformVersion: Platform.Version,
+    network: network.toUpperCase(), // Stellar network (TESTNET, PUBLIC, FUTURENET)
+    appVersion: getVersion(),
+    buildVersion: getBuildNumber(),
+    bundleId: getBundleId(),
+  };
+
+  // If analytics are disabled and we should respect that preference, return minimal context
+  if (respectAnalyticsPreference && !analyticsEnabled) {
+    return baseContext;
+  }
+
+  // Full context when analytics are enabled or when explicitly requested
+  const fullContext: Record<string, unknown> = {
+    ...baseContext,
+    publicKey: account?.publicKey ?? "N/A",
+    connectionType, // Internet connectivity (wifi, cellular, etc.)
+  };
+
+  // Add effectiveType only when available (mainly for cellular connections)
+  if (effectiveType) {
+    fullContext.effectiveType = effectiveType;
+  }
+
+  return fullContext;
+};
+
+/**
+ * Sets up Sentry context based on current app state and user preferences
+ */
+export const updateSentryContext = async (): Promise<void> => {
+  try {
+    const context = buildSentryContext();
+
+    Sentry.setContext("appContext", context);
```

Comment:
does it matter if we are calling `Sentry.setContext` before or after `Sentry.setUser` ?